File: dataset/ZeroShot/CLAUDE_claude-sonnet-4-5-20250929/Task168.py

def tweak_letters(s, tweaks):
    if s is None or tweaks is None or len(s) != len(tweaks):
        return s
    
    result = []
    
    for i in range(len(s)):
        current_char = s[i]
        tweak = tweaks[i]
        
        # Apply the tweak to the character
        new_char_code = ord(current_char) + tweak
        
        # Handle wrapping for lowercase letters
        if 'a' <= current_char <= 'z':
            if new_char_code > ord('z'):
                new_char_code = ord('a') + (new_char_code - ord('z') - 1)
            elif new_char_code < ord('a'):
                new_char_code = ord('z') - (ord('a') - new_char_code - 1)
        
        result.append(chr(new_char_code))
    
    return ''.join(result)


if __name__ == "__main__":
    # Test case 1
    print(tweak_letters("apple", [0, 1, -1, 0, -1]))
    # Expected: "aqold"
    
    # Test case 2
    print(tweak_letters("many", [0, 0, 0, -1]))
    # Expected: "manx"
    
    # Test case 3
    print(tweak_letters("rhino", [1, 1, 1, 1, 1]))
    # Expected: "sijop"
    
    # Test case 4
    print(tweak_letters("zebra", [1, 0, 0, 0, -1]))
    # Expected: "zebrq"
    
    # Test case 5
    print(tweak_letters("hello", [0, 0, 0, 0, 0]))
    # Expected: "hello"

File: dataset/ZeroShot/CLAUDE_claude-sonnet-4-5-20250929/Task168.java

public class Task168 {
    public static String tweakLetters(String str, int[] tweaks) {
        if (str == null || tweaks == null || str.length() != tweaks.length) {
            return str;
        }
        
        StringBuilder result = new StringBuilder();
        
        for (int i = 0; i < str.length(); i++) {
            char currentChar = str.charAt(i);
            int tweak = tweaks[i];
            
            // Apply the tweak to the character
            char newChar = (char) (currentChar + tweak);
            
            // Handle wrapping for lowercase letters
            if (currentChar >= 'a' && currentChar <= 'z') {
                if (newChar > 'z') {
                    newChar = (char) ('a' + (newChar - 'z' - 1));
                } else if (newChar < 'a') {
                    newChar = (char) ('z' - ('a' - newChar - 1));
                }
            }
            
            result.append(newChar);
        }
        
        return result.toString();
    }
    
    public static void main(String[] args) {
        // Test case 1
        System.out.println(tweakLetters("apple", new int[]{0, 1, -1, 0, -1}));
        // Expected: "aqold"
        
        // Test case 2
        System.out.println(tweakLetters("many", new int[]{0, 0, 0, -1}));
        // Expected: "manx"
        
        // Test case 3
        System.out.println(tweakLetters("rhino", new int[]{1, 1, 1, 1, 1}));
        // Expected: "sijop"
        
        // Test case 4
        System.out.println(tweakLetters("zebra", new int[]{1, 0, 0, 0, -1}));
        // Expected: "zebrq" (z+1 wraps to a, a-1 wraps to z)
        
        // Test case 5
        System.out.println(tweakLetters("hello", new int[]{0, 0, 0, 0, 0}));
        // Expected: "hello"
    }
}
